devolving_music/lib/song_scores.py

```diff
@@ -141,6 +147,35 @@ def get_info_sort(
         # return list of keys of dictionary of song objects sorted by info
         return info_submissions
 
+    @staticmethod
+    def weighted_lowest_info(
+            info_list: List["ScoreSuite"],
+            luck_factor=1) -> "ScoreSuite":
+        # luck factor greater than 1 means everything has a chance of being chosen
+        # luck factor less than 1 means that some higher informed songs will
+        # never be chosen
+        ceiling = round((info_list[-1].info_score) * luck_factor)
```

Comment:
This seems to depend on `info_list` being sorted a particular way. The argument just being named `info_list` certainly doesn't convey to another programmer that they are supposed to supply a sorted list.

I think you should move the `SongScores.get_info_sort` call from the song pair view into the body of this method so that there's no chance of accidentally calling this method on an unsorted list.

Reply:
This is fair I'll do this.